git-worktree/tests/copy_index/mod.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{dir_structure, fixture_path};
+use git_object::bstr::ByteSlice;
+use git_odb::FindExt;
+use git_worktree::{copy_index, Options};
+use std::fs;
+
+#[cfg(unix)]
+use std::os::unix::prelude::MetadataExt;
+
+type Result<T = ()> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+#[test]
+fn test_copy_index() -> Result<()> {
+    let path = fixture_path("make_repo");
+    let path_git = path.join(".git");
+    let mut file = git_index::File::at(path_git.join("index"), git_index::decode::Options::default())?;
+    let output_dir = tempfile::tempdir()?;
+    let output = output_dir.path();
+    let odb_handle = git_odb::at(path_git.join("objects")).unwrap();
+
+    copy_index(
+        &mut file,
+        &output,
+        move |oid, buf| {
+            odb_handle
+                .find(oid, buf)
+                .ok()
+                .map(|x| git_object::BlobRef::from_bytes(x.data).unwrap())
+        },
+        Options::default(),
+    )?;
+
+    let repo_files = dir_structure(&path);
+    let copy_files = dir_structure(output);
+
+    let srepo_files: Vec<_> = repo_files.iter().flat_map(|p| p.strip_prefix(&path)).collect();
+    let scopy_files: Vec<_> = copy_files.iter().flat_map(|p| p.strip_prefix(output)).collect();
+    assert!(srepo_files == scopy_files);
+
+    for (file1, file2) in repo_files.iter().zip(copy_files.iter()) {
+        assert!(fs::read(file1)? == fs::read(file2)?);
+        #[cfg(unix)]
+        assert!(fs::symlink_metadata(file1)?.mode() & 0b111 << 6 == fs::symlink_metadata(file2)?.mode() & 0b111 << 6);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_copy_index_without_symlinks() -> Result<()> {
+    let path = fixture_path("make_repo");
+    let path_git = path.join(".git");
+    let mut file = git_index::File::at(path_git.join("index"), git_index::decode::Options::default())?;
+    let output_dir = tempfile::tempdir()?;
+    let output = output_dir.path();
+    let odb_handle = git_odb::at(path_git.join("objects")).unwrap();
```

Comment:
If a test returns `crate::Result`, there should be no `unwrap()` anymore. Using `crate::Result` is only recommended at the very end of the development cycle when it's all working and stable, as it's generally harder to see where it breaks then, compared to a panic using `unwrap()`. Probably `anyhow` is better in that regard.

It's generally OK to use `unwrap()` in tests while having it return `()`, as in no explicit return value, if that makes debugging easier.